### ISSUE TITLE
Federation of (predefined) processes

### ISSUE THREAD

**Issue by soxofaan** [Open-EO/openeo-aggregator#4]
The aggregator combines multiple back-ends, each of which support different sets of predefined processes (or processes with slightly differing process definitions), as exposed by their respective `GET /processes` listings.

The aggregator's `/processes` currently returns the **intersection** of the process sets, because this is the safest choice with lowest surprise factor. 
An alternative is taking the **union**, which could cause problems that violate the openEO API: aggregator claims to support process X (from backend A), but execution (on backend B) fails with "unsupported process X" .

Note that even if the `/processes` just list the intersection of processes, a savvy user can still submit process graphs that use back-end specific processes if they know that the execution will happen on that back-end (e.g. because of which collections are used, or some other mechanism).

This ticket is for tracking questions and concerns about the strategy of process federation.

**Comment by m-mohr** [comment]
The more I'm working with Platform, the more the intersection of the processes is stopping me from succeeding with the Platform as the clients tell me that x or y is not available (although it is at one back-end), we also already got a RID based on something is not shown on <https://docs.openeo.cloud/processes>.
I think we should strongly think about using the union here.

**Comment by jdries** [comment]
Yes, I'm starting to feel the same. Even if we get all backends to support a core set of processes, it's still annoying for users that they have to switch.
Would it be an option that the aggregator can use an additional field to indicate which backends support a given process? Than maybe the web editor can apply some intuitive color coding, or small indication to help the user get a quick overview?

**Comment by soxofaan** [comment]
aggregator now returns union (initial implementation)

still to do / to discuss:
- how to handle differences in parameters/return (now VITO version has higher priority)
- annotate processes on which back-ends they are supported

**Comment by JeroenVerstraelen** [comment]
I have listed all discrepancies in process definitions among openeo backends here:
https://github.com/openEOPlatform/architecture-docs/issues/265

**Comment by soxofaan** [comment]
> I have listed all discrepancies in process definitions among openeo backends here: https://github.com/openEOPlatform/architecture-docs/issues/265

next iteration of this report: https://github.com/openEOPlatform/architecture-docs/issues/282

**Comment by soxofaan** [comment]
TODO: create new iteration of this report and fix low hanging fruit

**Comment by jdries** [comment]
Federation works, script to report discrepancies is available.
Upcoming compliancy test suite can/should also check process metadata.
